File: Patterns/Template Pattern/CSharp/BeverageTestDrive.cs
using System;

using Starbuzz;

public class BeverageTestDrive
{
    public static void Main(String[] args)
    {
        Starbuzz.TeaWithHook teaHook = new TeaWithHook();
        Starbuzz.CoffeeWithHook coffeeHook = new CoffeeWithHook();

        Console.WriteLine("\n홍차를 준비 중...");
        teaHook.PrepareRecipe();

        Console.WriteLine("\n커피를 준비 중...");
        coffeeHook.PrepareRecipe();
    }
}

File: Patterns/Template Pattern/Cpp/BeverageTestDrive.cpp
#include "Starbuzz/CoffeeWithHook.cpp"
#include "Starbuzz/TeaWithHook.cpp"
#include "Starbuzz/CaffeineBeverageWithHook.cpp"

using namespace std;

int main() {
    TeaWithHook* teaHook = new TeaWithHook();
    CoffeeWithHook* coffeeHook = new CoffeeWithHook();

    cout << "\n홍차 준비 중...\n";
    teaHook->prepareRecipe();

    cout << "\n커피 준비 중...\n";
    coffeeHook->prepareRecipe();
}